autosklearn/pipeline/components/data_preprocessing/feature_type.py

```diff
@@ -237,3 +237,23 @@ def _get_hyperparameter_search_space_recursevely(
                 return FeatTypeSplit._get_hyperparameter_search_space_recursevely(
                     dataset_properties, cs, st_operation)
         return cs
+
+    def to_sklearn(self) -> sklearn.base.BaseEstimator:
+        original_column_transformer = cast(
+            sklearn.compose.ColumnTransformer,
+            self.column_transformer,
+        )
```

Comment:
Wondering why a cast is needed considering it's constructed as one?

Reply:
The attribute `self.column_transformer` is initialized as `None`. Maybe we should initialize `self.column_transformer_` during `fit`?

Reply:
Ahh so is it the typical problem we find? 

```python
def __init__(self, ...):
   self.item : Optional[T] = None
   
def fit(self, ...):
   self.item = ...
   
def to_sklearn(self, ... ) -> T:
   return T  # Mypy says it could still be None
```

There's no real way around this as sklearn with its `__init__` and `fit` is kind of like an object that's constructed in two steps. I'm not the biggest fan of it but it's how it is.

If it's just checking which of the types `Union[T, None]` it is, I would rather an `assert obj is not None` instead of a `cast`.